src/intrinsics/ecma262/ArrayPrototype.js

```diff
@@ -980,6 +980,29 @@ export default function(realm: Realm, obj: ObjectValue): void {
     // 1. Let O be ? ToObject(this value).
     let O = To.ToObject(realm, context);
 
+    // if there is an initialValue, then the call will likely
+    // have side-effects so this is not supported for now
+    if (ArrayValue.isIntrinsicAndHasWidenedNumericProperty(O)) {
+      if (initialValue || initialValue instanceof UndefinedValue) {
```

Comment:
I don't get this condition. Did you mean:
if (initialValue && !(initialValue instanceof UndefinedValue)) {

Reply:
It's for when the value doesn't exist or has been passed in as undefined.

Reply:
Sorry, you were right with your first reply. Changed applied.